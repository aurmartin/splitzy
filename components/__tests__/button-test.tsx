import * as React from "react";
import { render, fireEvent } from "@testing-library/react-native";

import Button from "@/components/button";

it("renders correctly", () => {
  const tree = render(<Button>Press me!</Button>);
  expect(tree).toMatchSnapshot();
});

it("renders correctly with type", () => {
  const tree = render(<Button type="primary">Press me!</Button>);
  expect(tree).toMatchSnapshot();
});

it("renders correctly with style", () => {
  const tree = render(
    <Button style={{ backgroundColor: "red" }}>Press me!</Button>,
  );
  expect(tree).toMatchSnapshot();
});

it("handles onPress events", () => {
  const onPressMock = jest.fn();
  const { getByText } = render(
    <Button onPress={onPressMock}>Press me!</Button>,
  );

  fireEvent.press(getByText("Press me!"));
  expect(onPressMock).toHaveBeenCalled();
});

it("renders disabled state correctly", () => {
  const onPressMock = jest.fn();
  const tree = render(
    <Button disabled onPress={onPressMock}>
      Press me!
    </Button>,
  );
  expect(tree).toMatchSnapshot();
});
