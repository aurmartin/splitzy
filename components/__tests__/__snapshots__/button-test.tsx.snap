// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  enabled={true}
  hitSlop={
    {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  rippleColor={4290302944}
  style={
    [
      {},
      [
        {
          "alignItems": "center",
          "borderRadius": 40,
          "height": 40,
          "justifyContent": "center",
          "paddingHorizontal": 16,
        },
        {
          "backgroundColor": "hsl(200 40% 50%)",
        },
        undefined,
      ],
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Press me!
  </Text>
</RNGestureHandlerButton>
`;

exports[`renders correctly with style 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  enabled={true}
  hitSlop={
    {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  rippleColor={4290302944}
  style={
    [
      {},
      [
        {
          "alignItems": "center",
          "borderRadius": 40,
          "height": 40,
          "justifyContent": "center",
          "paddingHorizontal": 16,
        },
        {
          "backgroundColor": "hsl(200 40% 50%)",
        },
        {
          "backgroundColor": "red",
        },
      ],
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Press me!
  </Text>
</RNGestureHandlerButton>
`;

exports[`renders correctly with type 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  enabled={true}
  hitSlop={
    {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  rippleColor={4290302944}
  style={
    [
      {},
      [
        {
          "alignItems": "center",
          "borderRadius": 40,
          "height": 40,
          "justifyContent": "center",
          "paddingHorizontal": 16,
        },
        {
          "backgroundColor": "hsl(200 40% 50%)",
        },
        undefined,
      ],
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Press me!
  </Text>
</RNGestureHandlerButton>
`;

exports[`renders disabled state correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  enabled={false}
  hitSlop={
    {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  rippleColor={4290302944}
  style={
    [
      {},
      [
        {
          "alignItems": "center",
          "borderRadius": 40,
          "height": 40,
          "justifyContent": "center",
          "paddingHorizontal": 16,
        },
        {
          "backgroundColor": "hsl(200 40% 50%)",
        },
        undefined,
      ],
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Press me!
  </Text>
</RNGestureHandlerButton>
`;
